### PR TITLE
JAMES-3775 RspamdTask - handler rspamd timeout

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

File: third-party/rspamd/src/main/java/org/apache/james/rspamd/task/FeedHamToRspamdTaskDTO.java [hunk 1]
```diff
@@ -68,16 +69,20 @@ public static TaskDTOModule<FeedHamToRspamdTask, FeedHamToRspamdTaskDTO> module(
     private final double samplingProbability;
     private final Optional<Boolean> classifiedAsSpam;
 
+    private final int rspamdTimeout;
```
Comment (chibenwa) [inline third-party/rspamd/src/main/java/org/apache/james/rspamd/task/FeedHamToRspamdTaskDTO.java:72]: Please put the unit

```suggestion
    private final int rspamdTimeoutInSeconds;
```

File: third-party/rspamd/src/main/java/org/apache/james/rspamd/task/RunningOptions.java [hunk 2]
```diff
@@ -60,26 +60,37 @@ public boolean test(MessageResult messageResult) {
...
     public static final double DEFAULT_SAMPLING_PROBABILITY = 1;
     public static final Optional<Boolean> ALL_MESSAGES = Optional.empty();
-    public static final RunningOptions DEFAULT = new RunningOptions(DEFAULT_PERIOD, DEFAULT_MESSAGES_PER_SECOND, DEFAULT_SAMPLING_PROBABILITY, ALL_MESSAGES);
+    public static final RunningOptions DEFAULT = new RunningOptions(DEFAULT_PERIOD, DEFAULT_MESSAGES_PER_SECOND, DEFAULT_SAMPLING_PROBABILITY, ALL_MESSAGES, DEFAULT_RSPAMD_TIMEOUT);
 
     private final Optional<Long> periodInSecond;
     private final int messagesPerSecond;
     private final double samplingProbability;
     private final Optional<Boolean> classifiedAsSpam;
+    private final Integer rspamdTimeout;
```
Comment (chibenwa) [inline third-party/rspamd/src/main/java/org/apache/james/rspamd/task/RunningOptions.java:74]:
```suggestion
    private final Duration rspamdTimeout;
```

File: third-party/rspamd/src/main/java/org/apache/james/rspamd/route/FeedMessageRoute.java [hunk 3]
```diff
@@ -100,7 +101,8 @@ private RunningOptions getRunningOptions(Request request) {
         int messagesPerSecond = getMessagesPerSecond(request).orElse(RunningOptions.DEFAULT_MESSAGES_PER_SECOND);
         double samplingProbability = getSamplingProbability(request).orElse(RunningOptions.DEFAULT_SAMPLING_PROBABILITY);
         Optional<Boolean> classifiedAsSpam = getClassifiedAsSpam(request);
-        return new RunningOptions(periodInSecond, messagesPerSecond, samplingProbability, classifiedAsSpam);
+        int rspamdTimeout = ParametersExtractor.extractPositiveInteger(request, "rspamdTimeout").orElse(RunningOptions.DEFAULT_RSPAMD_TIMEOUT);
```
Comment (chibenwa) [inline third-party/rspamd/src/main/java/org/apache/james/rspamd/route/FeedMessageRoute.java:104]: Use DurationParser

**Review by chibenwa — COMMENTED** [review]
I do not see documentation for this parameter...

**Review by quantranhong1999 — COMMENTED** [review]
Nothing to add. Good to me.

**Review by chibenwa — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**